blockchain/blockchain.go

```diff
@@ -761,87 +714,6 @@ func (bc *blockchain) commitBlock(blk *block.Block) error {
 	return nil
 }
 
-func (bc *blockchain) pickAndRunActions(ctx context.Context, actionMap map[string][]action.SealedEnvelope,
-	ws factory.WorkingSet) ([]*action.Receipt, []action.SealedEnvelope, error) {
-	if bc.sf == nil {
-		return nil, nil, errors.New("statefactory cannot be nil")
-	}
-
-	receipts := make([]*action.Receipt, 0)
-	executedActions := make([]action.SealedEnvelope, 0)
-
-	bcCtx := protocol.MustGetBlockchainCtx(ctx)
-	blkCtx := protocol.MustGetBlockCtx(ctx)
-	registry := bcCtx.Registry
-	for _, p := range registry.All() {
-		if pp, ok := p.(protocol.PreStatesCreator); ok {
-			if err := pp.CreatePreStates(ctx, ws); err != nil {
-				return nil, nil, err
-			}
-		}
-	}
-
-	// initial action iterator
-	actionIterator := actioniterator.NewActionIterator(actionMap)
-	for {
-		nextAction, ok := actionIterator.Next()
-		if !ok {
-			break
-		}
-		receipt, err := ws.RunAction(ctx, nextAction)
-		if err != nil {
-			if errors.Cause(err) == action.ErrHitGasLimit {
-				// hit block gas limit, we should not process actions belong to this user anymore since we
-				// need monotonically increasing nounce. But we can continue processing other actions
-				// that belong other users
-				actionIterator.PopAccount()
-				continue
-			}
-			return nil, nil, errors.Wrapf(err, "Failed to update state changes for selp %x", nextAction.Hash())
-		}
-		if receipt != nil {
-			blkCtx.GasLimit -= receipt.GasConsumed
-			ctx = protocol.WithBlockCtx(ctx, blkCtx)
-			receipts = append(receipts, receipt)
-		}
-		executedActions = append(executedActions, nextAction)
-
-		// To prevent loop all actions in act_pool, we stop processing action when remaining gas is below
-		// than certain threshold
-		if blkCtx.GasLimit < bc.config.Chain.AllowedBlockGasResidue {
-			break
-		}
-	}
-	sk := bc.config.ProducerPrivateKey()
-	for _, p := range registry.All() {
-		if psac, ok := p.(protocol.PostSystemActionsCreator); ok {
-			elps, err := psac.CreatePostSystemActions(ctx)
-			if err != nil {
-				return nil, nil, err
-			}
-			for _, elp := range elps {
-				se, err := action.Sign(elp, sk)
-				if err != nil {
-					return nil, nil, err
-				}
```

Comment:
we need to keep this part in blockchain, that we don't need to pass bc.config into factory

Reply:
fixed in latest commit